roles/argocd_config/tasks/config-repo.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Validate Repo without SSH key
+  ansible.builtin.assert:
+    that:
+      - ac_repo | regex_search('^http[s]?://')
+      - ac_gh_token is defined
+      - ac_gh_username is defined
+    fail_msg:
+      - "Repository URL must start with 'http://' or 'https://'."
+      - "GitHub token and username must be defined when SSH key is not provided."
+  when:
+    - ac_ssh_key is not defined
+
+- name: Validate Repo with SSH key
+  ansible.builtin.assert:
+    that:
+      - ac_repo | regex_search('^\w+@')
+      - ac_gh_token is undefined
+      - ac_gh_username is undefined
+    fail_msg:
+      - "Repository URL must start with '<user>@' for SSH access, e.g., git@repo..."
+      - "GitHub token and username must not be defined when SSH key is provided."
+  when:
+    - ac_ssh_key is defined
+
+- name: Create temporary directory
+  ansible.builtin.tempfile:
+    state: directory
+    prefix: argocd_config_dir.
+  register: _ac_tmp_dir
+
+- name: Download ArgoCD CLI
+  ansible.builtin.get_url:
+    url: "https://github.com/argoproj/argo-cd/releases/latest/download/argocd-linux-amd64"
+    dest: "{{ _ac_tmp_dir.path }}/argocd"
+    mode: "0750"
+
+- name: Get ArgoCD password
+  kubernetes.core.k8s_info:
+    kind: Secret
+    namespace: "{{ ac_namespace }}"
+    name: openshift-gitops-cluster
+  register: _ac_sec
+  no_log: "{{ ac_hide_secrets | bool }}"
+
+- name: Get ArgoCD url
+  kubernetes.core.k8s_info:
+    kind: Route
+    namespace: "{{ ac_namespace }}"
+    name: openshift-gitops-server
+  register: _ac_route
+
+- name: Add repo to ArgoCD
```

Comment:
The handler may not run if the playbook fails before it is executed. Consider using a block for this task with an always: section to delete the temp directory